Source/WebCore/PAL/pal/text/TextCodecCJK.cpp

```diff
@@ -876,15 +876,15 @@ static uint32_t gb18030RangesPointer(char32_t codePoint)
 {
     if (codePoint == 0xE7C7)
         return 7457;
-    auto upperBound = std::ranges::upper_bound(gb18030Ranges(), makeSecondAdapter(codePoint), CompareSecond { });
-    ASSERT(upperBound != gb18030Ranges().begin());
-    uint32_t pointerOffset = (upperBound - 1)->first;
-    char32_t offset = (upperBound - 1)->second;
+    auto& ranges = gb18030Ranges();
+    auto upperBound = std::ranges::upper_bound(ranges, makeSecondAdapter(codePoint), CompareSecond { });
+    ASSERT(upperBound != ranges.begin());
+    auto& range = ranges[upperBound - ranges.begin() - 1];
+    uint32_t pointerOffset = range.first;
+    char32_t offset = range.second;
```

Comment:
Does something like this work?
```suggestion
    auto [pointerOffset, offset] = ranges[upperBound - ranges.begin() - 1];
```